File: 2)DIT/High_DIT.java
// DIT Level 0: Root Component
class BaseService {
    public void log(String message) {
        System.out.println("[BaseService] " + message);
    }
}

// DIT Level 1
class NetworkService extends BaseService {
    public String request(String url) {
        log("Requesting: " + url);
        return "response";
    }
}

// DIT Level 2
class AuthService extends NetworkService {
    public boolean authenticate(String token) {
        log("Authenticating token...");
        return token.equals("valid");
    }
}

// DIT Level 3
class UserService extends AuthService {
    public User fetchUser(int id) {
        log("Fetching user...");
        return new User(id, "Rithika", "Admin");
    }
}

// Helper User model
class User {
    int id;
    String name;
    String role;

    public User(int id, String name, String role) {
        this.id = id;
        this.name = name;
        this.role = role;
    }
}

// DIT Level 4 → High DIT Component
class UserProfileManager extends UserService {
    public User loadProfile(int id, String token) {
        if (!authenticate(token)) {
            log("Auth failed!");
            return null;
        }

        User user = fetchUser(id);
        log("Loaded Profile → " + user.name);
        return user;
    }
}

// Usage
public class High_DIT {
    public static void main(String[] args) {
        UserProfileManager manager = new UserProfileManager();
        manager.loadProfile(101, "valid");
    }
}

File: 7)Ck_Analyzer/High_DIT.ts
// DIT Level 0: Root Component
class BaseService {
    log(message: string) { console.log("[BaseService]", message); }
}

// DIT Level 1
class NetworkService extends BaseService {
    request(url: string) {
        this.log("Requesting: " + url);
        return { data: "response" };
    }
}

// DIT Level 2
class AuthService extends NetworkService {
    authenticate(valid_token: string) {
        this.log("Authenticating token...");
        return valid_token === "valid";
    }
}

// DIT Level 3
class UserService extends AuthService {
    fetchUser(id: number) {
        this.log("Fetching user...");
        return { id, name: "Rithika", role: "Admin" };
    }
}

// DIT Level 4 → HIGH DIT Component
class UserProfileManager extends UserService {
    loadProfile(id: number, valid_token: string) {
        if (!this.authenticate(valid_token)) {
            this.log("Auth failed!");
            return null;
        }

        const user = this.fetchUser(id);
        this.log("Loaded Profile → " + user.name);
        return user;
    }
}

// Usage
const manager = new UserProfileManager();
manager.loadProfile(101, "valid");
